common/types.go

```diff
@@ -86,8 +86,34 @@ type ReadParams struct {
 	Since time.Time // optional, omit this to fetch all records
 	// Deleted is true if we want to read deleted records instead of active records.
 	Deleted bool // optional, defaults to false
+	// The size of the returned records per page
```

Comment:
Still thinking about this as we need to be cognizant of other providers while doing this cc (filters look different from provider to provider, for example) @laurenzlong @eberle1080

Reply:
While these are really good ideas, generally we don't introduce new capabilities to connectors unless we are confident we can support it across **all** connectors now and in the future. While Outreach has support for custom page sizes, filters and sorting, a lot of other APIs won't, and it would be really unexpected behavior if these fields of ReadParams sometimes works and sometimes does nothing. While there are things we can do to "fake it" for providers that don't support these capabilities natively, it adds complexity and we would be committing to taking on that work for future connectors. So can you please remove this PageSize, FilterBy and SortBy, and remove the relevant code? Sorry to make you take out code that you've already written but we just have to be really thoughtful about future maintenance of this project.

Reply:
Noted Lauren